src/test/scala/is/hail/utils/IntervalSuite.scala

```diff
@@ -0,0 +1,211 @@
+package is.hail.utils
+
+import is.hail.annotations.ExtendedOrdering
+import is.hail.expr.types.TInt32
+import org.testng.annotations.Test
+
+class IntervalSuite {
+
+  val pord: ExtendedOrdering = TInt32().ordering
+
+  // set of intervals chosen from 5 endpoints spans the space of relations
+  // that two non-empty intervals can have with each other.
+  val points: IndexedSeq[Int] = 1 to 5
+
+  val test_intervals: IndexedSeq[SetInterval] =
+    for {
+      s <- points
+      e <- points
+      is <- Array(true, false)
+      ie <- Array(true, false)
+    } yield SetInterval(s, e, is, ie)
+
+  val test_itrees: IndexedSeq[SetIntervalTree] =
+    for {
+      i1 <- test_intervals
+      i2 <- test_intervals
+    } yield SetIntervalTree(Array(i1, i2).zipWithIndex)
+
+  def shouldOrShouldnt(bool: Boolean): String = if (bool) "should" else "shouldn't"
+
+  @Test def intervalEmpty() {
```

Comment:
what if we broke with scala convention even more and just used underscores? underscores are nice.

Reply:
I almost suggested this.  I strongly agree but it will lead to some extreme inconsistency for a while.  (Do we just do it in test names?  All new code, everywhere?)